mlflow/store/rest_store.py

```diff
@@ -121,6 +121,12 @@ def get_experiment_by_name(self, name):
         response_proto = self._call_endpoint(GetExperimentByName, req_body)
         return Experiment.from_proto(response_proto.experiment)
 
+    def delete_experiment(self, experiment_id):
```

Comment:
Let's hold off on adding experiment deletion APIs in this PR - @mparkhe is working on experiment deletion if you're interested in collaborating / reviewing :)